src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    }
+
+    let result = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || typeof this[i] === 'undefined') {
+        result += '' + String(separator);
+      } else {
+        result += this[i] + String(separator);
+      }
+    }
```

Comment:
you can create a separate variable for the value (String(separator)), to not repeat yourself)

Reply:
done